### PR TITLE
[GEOT-6716]: Support COG on ImageMosaic

### PR DESCRIPTION
**Pull request by dromagnoli**
Note that:
The PR contains a temporary pom.xml update to use imageio-ext snapshot.
The final PR will be updated to use a newly released imageio-ext 1.3.X version.

The PR contains GEOT-6715 for reviewing purposes.

This PR addresses several feedback from the previous PR which I'm closing:
https://github.com/geotools/geotools/pull/3172

### CODE REVIEW

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/ImageMosaicDatastoreWalker.java [hunk 1]
```diff
@@ -233,4 +257,151 @@ public void run() {
             }
         }
     }
+
+    protected void handleCogGranule(final String granuleBeingProcessed) throws IOException {
```
Comment (aaime) [inline modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/ImageMosaicDatastoreWalker.java:261]: So in the end the handling of COG refs and normal files was too different to homogeneize eh?

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/catalog/CatalogConfigurationBean.java [hunk 2]
```diff
@@ -46,6 +46,11 @@
 
     private boolean heterogeneousCRS;
 
+    /** <code>true</code> if the mosaic is made of remote COG granules. */
+    private boolean cog;
```
[the remark this answers is not in the document]
Comment (dromagnoli) [reply modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/catalog/CatalogConfigurationBean.java:50]: I had that idea indeed at some point but I left the boolean for "lazyness" check (isCog vs getCogBean()!=null)  on the various places around on property parsing/loading.
Anyway, I can remove it.

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/ImageMosaicDatastoreWalker.java (hunk 1 above)
Comment (dromagnoli) [reply modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/ImageMosaicDatastoreWalker.java:261]: yep. there was several file-centric checks and assumptions here and there.

**Comment by aaime** [comment]
@dromagnoli there are 3 PMD violations in the mosaic module:

```
[INFO] --- maven-pmd-plugin:3.12.0:check (default) @ gt-imagemosaic ---
[INFO] PMD Failure: org.geotools.gce.imagemosaic.CogGranuleAccessProvider$CogGranuleAccessProvider:56 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'LOGGER'..
[INFO] PMD Failure: org.geotools.gce.imagemosaic.GranuleDescriptor$GranuleDescriptor:365 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'layout'..
[INFO] PMD Failure: org.geotools.gce.imagemosaic.catalog.CatalogConfigurationBean$CatalogConfigurationBean:50 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'cog'..
```

**Comment by dromagnoli** [comment]
> @dromagnoli there are 3 PMD violations in the mosaic module:
> 
> ```
> [INFO] --- maven-pmd-plugin:3.12.0:check (default) @ gt-imagemosaic ---
> [INFO] PMD Failure: org.geotools.gce.imagemosaic.CogGranuleAccessProvider$CogGranuleAccessProvider:56 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'LOGGER'..
> [INFO] PMD Failure: org.geotools.gce.imagemosaic.GranuleDescriptor$GranuleDescriptor:365 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'layout'..
> [INFO] PMD Failure: org.geotools.gce.imagemosaic.catalog.CatalogConfigurationBean$CatalogConfigurationBean:50 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'cog'..
> ```

Thanks for the report.
PR Updated. 
it should be fixed now.

**Pull request merged**